### PR TITLE
feat(registry): features require a secure context by default, with opt-out

### PR DESCRIPTION
**Pull request by jyasskin**
By default, features will always be "denied" in non-secure contexts. The
"allowed in non-secure contexts" flag undoes this default.

It would be possible to throw an exception from request() and possibly
query() when a secure-only feature is used from a non-secure context,
but denying leads to one fewer behavior that developers need to account
for.

Preview at https://api.csswg.org/bikeshed/?url=https://raw.githubusercontent.com/jyasskin/permissions/secure-permissions/index.bs#requires-a-secure-context

@mikewest

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -350,9 +354,15 @@ spec: webidl
...
     Each enumeration value in the {{PermissionName}} enum identifies a
     <a>powerful feature</a>. Each <a>powerful feature</a> has the following
-    permission-related algorithms and types:
+    permission-related flags, algorithms, and types:
   </p>
   <dl>
+    <dt>A <dfn export>requires a secure context</dfn> flag</dt>
+    <dd>
+      Indicates that the feature will always be {{"denied"}} in <a>non-secure
+      contexts</a>. This flag defaults to un-set.
```
Comment (jyasskin) [inline index.bs]: Arguably, we should default this to "set", and make features explicitly say that they don't require a secure context... Then I'd probably reword it to reverse the sense: "allowed in non-secure contexts" defaulting again to un-set.

Thoughts?

Comment (martinthomson) [inline index.bs]: I would prefer to default to disabling these features in non-secure contexts.

Comment (jyasskin) [inline index.bs]: Done, thanks.

File: index.bs [hunk 2]
```diff
@@ -517,7 +530,8 @@ spec: webidl
       The <dfn>"camera"</dfn>, <dfn>"microphone"</dfn> , and
       <dfn>"speaker"</dfn>
       permissions are associated with permission to use media devices as
-      specified in [[GETUSERMEDIA]] and [[audio-output]].
+      specified in [[GETUSERMEDIA]] and [[audio-output]]. {{"speaker"}} is
```
Comment (jyasskin) [inline index.bs]: @alvestrand Did I get the right set of media-related permissions that require secure contexts?
